scaleway/resource_domain_record.go

```diff
@@ -0,0 +1,426 @@
+package scaleway
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	domain "github.com/scaleway/scaleway-sdk-go/api/domain/v2beta1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func resourceScalewayDomainRecord() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceScalewayDomainRecordCreate,
+		ReadContext:   resourceScalewayDomainRecordRead,
+		UpdateContext: resourceScalewayDomainRecordUpdate,
+		DeleteContext: resourceScalewayDomainRecordDelete,
+		Timeouts: &schema.ResourceTimeout{
+			Default: schema.DefaultTimeout(defaultDomainRecordTimeout),
+		},
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"dns_zone": {
+				Type:        schema.TypeString,
+				Description: "The zone you want to add the record in",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"keep_empty_zone": {
+				Type:        schema.TypeBool,
+				Description: "When destroy a ressource record, if a zone have only NS, delete the zone",
```

Comment:
`Description: "When destroy a resource record, if a zone have only NS, delete the zone",
`